polars/polars-io/src/utils.rs

```diff
@@ -16,7 +15,9 @@ use crate::ArrowSchema;
 pub fn resolve_homedir(path: &Path) -> PathBuf {
     // replace "~" with home directory
     if path.starts_with("~") {
-        if let Some(homedir) = home_dir() {
+        // home crate does not compile on wasm https://github.com/rust-lang/cargo/issues/12297
+        #[cfg(not(target_arch = "wasm32"))]
+        if let Some(homedir) = home::home_dir() {
             return homedir.join(path.strip_prefix("~").unwrap());
         }
     }
```

Comment:
this is probably fine if we are just concerned with _compiling_, but since WASM doesn't have a filesystem, i'm wondering if `resolve_homedir` should be feature flagged instead? Likely anything that calls `resolve_homedir` will panic on `wasm32` anyways.